File: apps/web/ce/types/worklog.ts
/**
 * Copyright (c) 2026-present Maciej Szymczak and contributors
 * SPDX-License-Identifier: AGPL-3.0-only
 * See the LICENSE file for details.
 */

export type TIssueWorklog = {
  id: string;
  duration: number; // minutes
  description: string;
  logged_by: string | null;
  issue: string;
  project: string;
  workspace: string;
  created_at: string;
  updated_at: string;
};

export type TIssueWorklogsResponse = {
  worklogs: TIssueWorklog[];
  total_duration: number; // minutes
};

File: apps/api/plane/ext/serializers/worklog.py
# Copyright (c) 2026-present Maciej Szymczak and contributors
# SPDX-License-Identifier: AGPL-3.0-only
# See the LICENSE file for details.

from rest_framework import serializers

from plane.ext.models import IssueWorkLog


class IssueWorkLogSerializer(serializers.ModelSerializer):
    class Meta:
        model = IssueWorkLog
        fields = [
            "id",
            "duration",
            "description",
            "logged_by",
            "issue",
            "project",
            "workspace",
            "created_at",
            "updated_at",
        ]
        read_only_fields = ["id", "logged_by", "issue", "project", "workspace", "created_at", "updated_at"]
